test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -87,13 +88,17 @@ private void assertCommandBehavior(String inputCommand,
             assertEquals(lastShownList, r.getRelevantPersons().get());
         }
 
+        //Obtain command type
+        Command comd = new Parser().parseCommand(inputCommand);
+
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if (comd.isMutating()) {
```

Comment:
To be complete, you should have an `else` branch that confirms the file was not updated for non mutating commands (e.g. by checking the last updated time of the file).